### PR TITLE
Map GPOS severity level to GPDB Severity Levels

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
GPOS raises exception with different severity level, but
they were being logged to GPDB logs at LOG severity level.
This disabled users to not turn off logging for GPOS exceptions, unless
GPDB log setting was changed higher than LOG severity level.

This is the initial commit which introduces the functionality. If an
exception is created without the GPDB severity level, it will default to
LOG severity level in GPDB.

Signed-off-by: Jemish Patel <jpatel@pivotal.io>

### CODE REVIEW

**Comment by vraghavan78** [comment]
@bhuvnesh2703 let us not target this for GPDB 4.3.17.0. Let us wait for the next 4.3 release along with 5.1

File: libgpos/src/error/CException.cpp [hunk 1]
```diff
@@ -143,6 +163,39 @@ CException::Raise
 }
 
 
+void
+CException::Raise
```
Comment (oarap) [inline libgpos/src/error/CException.cpp]: Is it possible to refactor? Both version of Raise seems to be sharing significant code.

Comment (bhuvnesh2703) [reply libgpos/src/error/CException.cpp]: i think we cannot, as VA_START excepts the functions from which it is called to be a variable length and in some cases we want to send ulMinor and in some cases ulSeverityLevel, so would prefer keep it this way. Also i believe in future once we change all the calls for other GPOS_RAISE we can remove this other function completely.

File: libgpos/include/gpos/error/CException.h [hunk 2]
```diff
@@ -22,6 +22,12 @@
 #define GPOS_RAISE(...) \
 	gpos::CException::Raise(__FILE__, __LINE__, __VA_ARGS__)
 
+// raises GPOS exception,
+// these exceptions can later be translated to GPDB log severity levels
+// so that they can be written in GPDB with appropriate severity level.
+#define GPOS_THROW_EXCEPTION(...) \
+gpos::CException::Raise(__FILE__, __LINE__, __VA_ARGS__)
```
Comment (dhanashreek89) [inline libgpos/include/gpos/error/CException.h]: Indentation is off here

**Pull request merged**